python/lsst/daf/butler/datastores/s3Datastore.py

```diff
@@ -212,8 +211,9 @@ def _write_in_memory_to_artifact(self, inMemoryDataset: Any, ref: DatasetRef) ->
             with tempfile.NamedTemporaryFile(suffix=formatter.extension) as tmpFile:
                 formatter._fileDescriptor.location = Location(*os.path.split(tmpFile.name))
                 formatter.write(inMemoryDataset)
-                self.client.upload_file(Bucket=location.netloc, Key=location.relativeToPathRoot,
-                                        Filename=tmpFile.name)
+                with open(tmpFile.name, 'rb') as f:
+                    self.client.put_object(Bucket=location.netloc,
```

Comment:
So to point out what I said in comments on the test utils call type changes. Changing from [`upload_file`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.upload_file) to [`put_objects`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.put_object) replaces the underlying high level call through S3Transfer class wiht a low level API call. 

Upload file call automatically multipart-uploads the file whereas put object does not. Additionally, if anyone wanted to ever implement a better, tighter upload control configuration this would now prevent them from doing so.

Reply:
The multipart upload issue was indeed the motivation behind this change (or call it workaround)